Steve Papacharalambous <stevep@freescale.com> 17 September 2008:
Patch to fix build errors when building with gcc-4.3.x and glibc-2.8.  Errors
are:

[snip]
if gcc -std=gnu99  -I. -I. -I.      -g -O2 -MT utimecmp.o -MD -MP -MF ".deps/utimecmp.Tpo" -c -o utimecmp.o utimecmp.c; \
        then mv -f ".deps/utimecmp.Tpo" ".deps/utimecmp.Po"; else rm -f ".deps/utimecmp.Tpo"; exit 1; fi
In file included from utimecmp.c:33:
utimens.h:2: error: conflicting types for 'futimens'
/opt/freescale/usr/local/gcc-4.3.8-eglibc-2.8.8/powerpc-linux-gnu/powerpc-linux-gnu/libc/usr/include/sys/stat.h:370: error: previous declaration of 'futimens' was here
make[2]: *** [utimecmp.o] Error 1
make[2]: Leaving directory `/home/stevep/work/8548/ltib/rpm/BUILD/coreutils-6.3/lib'
make[1]: *** [all] Error 2
[/snip]

Patch back ported from coreutils-6.10
==============================================================================
diff --exclude CVS -uNr coreutils-6.3/lib/utimens.c coreutils-6.3.modified/lib/utimens.c
--- coreutils-6.3/lib/utimens.c	Thu Sep 14 10:53:59 2006
+++ coreutils-6.3.modified/lib/utimens.c	Wed Sep 17 16:52:22 2008
@@ -73,7 +73,7 @@
    Return 0 on success, -1 (setting errno) on failure.  */
 
 int
-futimens (int fd ATTRIBUTE_UNUSED,
+gl_futimens (int fd ATTRIBUTE_UNUSED,
 	  char const *file, struct timespec const timespec[2])
 {
   /* There's currently no interface to set file timestamps with
@@ -166,5 +166,5 @@
 int
 utimens (char const *file, struct timespec const timespec[2])
 {
-  return futimens (-1, file, timespec);
+  return gl_futimens (-1, file, timespec);
 }
diff --exclude CVS -uNr coreutils-6.3/lib/utimens.h coreutils-6.3.modified/lib/utimens.h
--- coreutils-6.3/lib/utimens.h	Tue Nov 23 21:39:48 2004
+++ coreutils-6.3.modified/lib/utimens.h	Wed Sep 17 16:52:38 2008
@@ -1,3 +1,3 @@
 #include "timespec.h"
-int futimens (int, char const *, struct timespec const [2]);
+int gl_futimens (int, char const *, struct timespec const [2]);
 int utimens (char const *, struct timespec const [2]);
diff --exclude CVS -uNr coreutils-6.3/src/copy.c coreutils-6.3.modified/src/copy.c
--- coreutils-6.3/src/copy.c	Sun Sep 24 18:01:10 2006
+++ coreutils-6.3.modified/src/copy.c	Wed Sep 17 16:53:34 2008
@@ -491,7 +491,7 @@
       timespec[0] = get_stat_atime (src_sb);
       timespec[1] = get_stat_mtime (src_sb);
 
-      if (futimens (dest_desc, dst_name, timespec) != 0)
+      if (gl_futimens (dest_desc, dst_name, timespec) != 0)
 	{
 	  error (0, errno, _("preserving times for %s"), quote (dst_name));
 	  if (x->require_preserve)
diff --exclude CVS -uNr coreutils-6.3/src/touch.c coreutils-6.3.modified/src/touch.c
--- coreutils-6.3/src/touch.c	Wed Nov  2 10:01:07 2005
+++ coreutils-6.3.modified/src/touch.c	Wed Sep 17 16:53:05 2008
@@ -182,7 +182,7 @@
       t = timespec;
     }
 
-  ok = (futimens (fd, (fd == STDOUT_FILENO ? NULL : file), t) == 0);
+  ok = (gl_futimens (fd, (fd == STDOUT_FILENO ? NULL : file), t) == 0);
 
   if (fd == STDIN_FILENO)
     {
